### PR TITLE
[SQL] Updated SDK with correct API version

### PR DESCRIPTION
**Pull request by IliasKhan**
## Description

This API version in SDK was changed in [Commit](https://github.com/Azure/azure-powershell/commit/ec9684cc05ff91bab74e3f4e9b699415877f8c4a) which started the breakage in Build. Updated the sdk to fix the build.

I made the correct changes with correct api in my original PR 6 month back [[SQL] Updated devops audit api by IliasKhan · Pull Request #20447 · Azure/azure-powershell (github.com)](https://github.com/Azure/azure-powershell/pull/20447)..I generated the sdk this time only.

Later in https://github.com/Azure/azure-powershell/commit/ec9684cc05ff91bab74e3f4e9b699415877f8c4a commit api version of readme.md file got changed and for some reason. From then, sql build started failing.
 
This change will fix the sql build and update the readme file. sdk files related to this did not change may be because it was already changed in my original PR.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Please update the `ChangeLog.md` under your module.

File: src/Sql/Sql.Sdk/Generated/SdkInfo_SqlManagementClient.cs [hunk 1]
```diff
@@ -96,7 +96,7 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "ServerAzureADOnlyAuthentications", "2020-11-01-preview"),
                 new Tuple<string, string, string>("Sql", "ServerBlobAuditingPolicies", "2021-11-01-preview"),
                 new Tuple<string, string, string>("Sql", "ServerConfigurationOptions", "2022-08-01-preview"),
-                new Tuple<string, string, string>("Sql", "ServerDevOpsAuditSettings", "2020-11-01-preview"),
+                new Tuple<string, string, string>("Sql", "ServerDevOpsAuditSettings", "2022-05-01-preview"),
```
Comment (vidai-msft) [inline src/Sql/Sql.Sdk/Generated/SdkInfo_SqlManagementClient.cs:99]: We don't support updating the api-version directly from the generated sdk files. Please update the api-version from README.md and then re-generate SDK code.

File: src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs [hunk 2]
```diff
@@ -112,7 +112,7 @@ public ServerBlobAuditingPolicy()
         /// audit logs.
         ///
         /// For more information, see [Database-Level Audit Action
-        /// Groups](https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups).
+        /// Groups](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups).
```
Comment (vidai-msft) [inline src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs:115]: The docs URL has been updated to learn.microsoft.com instead of docs.microsoft.com. Please revert the change and re-generate the code.

File: src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs [hunk 3]
```diff
@@ -139,7 +139,7 @@ public ServerBlobAuditingPolicy()
         /// SELECT on SCHEMA::mySchema by public
         ///
         /// For more information, see [Database-Level Audit
-        /// Actions](https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)</param>
+        /// Actions](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)</param>
```
Comment (vidai-msft) [inline src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs:142]: The docs URL has been updated to learn.microsoft.com instead of docs.microsoft.com. Please revert the change and re-generate the code.

File: src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs [hunk 4]
```diff
@@ -293,7 +293,7 @@ public ServerBlobAuditingPolicy(BlobAuditingPolicyState state, string id = defau
         /// audit logs.
         ///
         /// For more information, see [Database-Level Audit Action
-        /// Groups](https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups).
+        /// Groups](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups).
```
Comment (vidai-msft) [inline src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs:296]: The docs URL has been updated to learn.microsoft.com instead of docs.microsoft.com. Please revert the change and re-generate the code.

File: src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs [hunk 5]
```diff
@@ -321,7 +321,7 @@ public ServerBlobAuditingPolicy(BlobAuditingPolicyState state, string id = defau
         /// SELECT on SCHEMA::mySchema by public
         ///
         /// For more information, see [Database-Level Audit
-        /// Actions](https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)
+        /// Actions](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions)
```
Comment (vidai-msft) [inline src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs:324]: The docs URL has been updated to learn.microsoft.com instead of docs.microsoft.com. Please revert the change and re-generate the code.

**Review by vidai-msft — CHANGES_REQUESTED** [review]

File: src/Sql/Sql.Sdk/Generated/SdkInfo_SqlManagementClient.cs (hunk 1 above)
Comment (IliasKhan) [reply src/Sql/Sql.Sdk/Generated/SdkInfo_SqlManagementClient.cs:99]: I did not update this manually;This file is generated. I updated the readme file. [here](https://github.com/Azure/azure-powershell/pull/21898/files#diff-c93755d6b1c06176d1b3246b84a795992e0495756584650faf37522e2f3149ba).

File: src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs (hunk 2 above)
Comment (IliasKhan) [reply src/Sql/Sql.Sdk/Generated/Models/ServerBlobAuditingPolicy.cs:115]: My commit was reverted in this [commit](https://github.com/Azure/azure-powershell/commit/ec9684cc05ff91bab74e3f4e9b699415877f8c4a). From here it started the breakage in the build

**Comment by IliasKhan** [comment]
> Please update the `ChangeLog.md` under your module.

Do I really need to add as this is internal build fix. And I think we publish the changelog comments.

**Comment by vidai-msft** [comment]
@IliasKhan can you please do the following?
1. Revert the change of ServerBlobAuditingPolicy.cs since the only change is replace "learn.microsoft.com" to "docs.microsoft.com" which is not necessary and wrong.
2. For the SDK generated code, I also noticed one file updated **PrivateEndpointConnectionProperties.cs** (one property **GroupId** is added) and one file added **ServerPublicNetworkAccessFlag.cs**. Can you confirm it won't cause any problem if they are not included in the PR?

**Comment by IliasKhan** [comment]
> @IliasKhan can you please do the following?
> 
> 1. Revert the change of ServerBlobAuditingPolicy.cs since the only change is replace "learn.microsoft.com" to "docs.microsoft.com" which is not necessary and wrong.
> 2. For the SDK generated code, I also noticed one file updated **PrivateEndpointConnectionProperties.cs** (one property **GroupId** is added) and one file added **ServerPublicNetworkAccessFlag.cs**. Can you confirm it won't cause any problem if they are not included in the PR?

Hi Removed the files with only comments change.
Now, if you try to generate sdk without my change, still those files will get generated. So, not committing other generated files will not create any issue.

**Review by vidai-msft — APPROVED** [review]
